pkg/cmd/admin/policy/remove_from_project.go

```diff
@@ -103,20 +105,25 @@ func (o *RemoveFromProjectOptions) Run() error {
 
 	usersRemoved := util.StringSet{}
 	groupsRemoved := util.StringSet{}
+	sasRemoved := util.StringSet{}
+	othersRemoved := util.StringSet{}
+
+	subjectsToRemove := authorizationapi.BuildSubjects(o.Users, o.Groups, uservalidation.ValidateUserName, uservalidation.ValidateGroupName)
 
 	for _, currPolicyBinding := range bindingList.Items {
 		for _, currBinding := range authorizationapi.SortRoleBindings(currPolicyBinding.RoleBindings, true) {
-			bindingHasGroups := len(o.Groups) > 0 && currBinding.Groups.HasAny(o.Groups...)
-			bindingHasUsers := len(o.Users) > 0 && currBinding.Users.HasAny(o.Users...)
-			if !bindingHasGroups && !bindingHasUsers {
-				continue
-			}
+			originalSubjects := make([]kapi.ObjectReference, len(currBinding.Subjects))
+			copy(originalSubjects, currBinding.Subjects)
+			oldUsers, oldGroups, oldSAs, oldOthers := authorizationapi.SubjectsStrings(currBinding.Namespace, originalSubjects)
+			oldUsersSet, oldGroupsSet, oldSAsSet, oldOtherSet := util.NewStringSet(oldUsers...), util.NewStringSet(oldGroups...), util.NewStringSet(oldSAs...), util.NewStringSet(oldOthers...)
 
-			existingGroups := util.NewStringSet(currBinding.Groups.List()...)
-			existingUsers := util.NewStringSet(currBinding.Users.List()...)
+			currBinding.Subjects = removeSubjects(currBinding.Subjects, subjectsToRemove)
+			newUsers, newGroups, newSAs, newOthers := authorizationapi.SubjectsStrings(currBinding.Namespace, currBinding.Subjects)
```

Comment:
At this point, we'd really benefit from []kapi.ObjectReference having a type alias that let you do Difference / Subset operations on it (without being converted to a map, because every single one of these is going to be a small list).  That would reduce the amount of redundant stuff you're doing here.  We don't really need be comparing the string sets (and that's basically transforming the type into a lesser type).  

I don't think you *have* to do it here, but I'd basically like to see us normalize []kapi.ObjectReference across the stack.